src/Workspaces/Core/Portable/Extensions/BasicBlockExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+    internal static partial class BasicBlockExtensions
+    {
+        public static IEnumerable<IOperation> DescendantOperations(this BasicBlock basicBlock)
+        {
+            foreach (var statement in basicBlock.Operations)
+            {
+                foreach (var operation in statement.DescendantsAndSelf())
+                {
+                    yield return operation;
+                }
+            }
+
+            if (basicBlock.BranchValue != null)
+            {
+                foreach (var operation in basicBlock.BranchValue.DescendantsAndSelf())
+                {
+                    yield return operation;
+                }
+            }
+        }
```

Comment:
this feels like:    `basicBlock.Operations.SelectMany(s => s.DescendentsAndSelf()).Concat(...)`

Reply:
`basicBlock.Operations` does not include BranchValue.

---
In reply to: [230606533](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606533) [](ancestors = 230606533)

Reply:
sorry, i should have been clearer.  i was just implying that hte 'BranchValue' part would go in the 'Concat' portion.  My overall point was that thsi was a lot of code for what seemed like a pretty simple linq-method-query.  Your call :)

Reply:
The branch value can be null, so there will need to be a non-null check as well. Keeping this as is for now.

---
In reply to: [230988933](https://github.com/dotnet/roslyn/pull/30777#discussion_r230988933) [](ancestors = 230988933)